### PR TITLE
Add sign in and sign up integration w/ keycloak

### PR DESCRIPTION
**Pull request by Cwighty**
can sign in with keycloak, creates user in db if new

### CODE REVIEW

File: nutrinova-api/.editorconfig [hunk 1]
```diff
@@ -110,4 +110,9 @@ dotnet_diagnostic.IDE0065.severity = none
 dotnet_diagnostic.IDE0009.severity = none
 dotnet_diagnostic.SA1101.severity = none
 # IDE0005: Remove unused usings
-dotnet_diagnostic.IDE0005.severity = error
\ No newline at end of file
+dotnet_diagnostic.IDE0005.severity = error
+
+[NutrinovaDbContext.cs]
+dotnet_diagnostic.CS8618.severity = none
+[NutrinovaDbContext.cs]
```
Comment (garv5014) [inline nutrinova-api/.editorconfig:117]: you can remove the second [NutrinovaDbContext.cs ]  I think

File: nutrinova-api/NutrinovaData/NutrinovaDbContext.cs [hunk 2]
```diff
@@ -49,72 +51,75 @@ public NutrinovaDbContext(DbContextOptions<NutrinovaDbContext> options)
     public virtual DbSet<Unit> Units { get; set; }
 
     protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
-
-        // #warning To protect potentially sensitive information in your connection string, you should move it out of source code. You can avoid scaffolding the connection string by using the Name= syntax to read it from configuration - see https://go.microsoft.com/fwlink/?linkid=2131148. For more guidance on storing connection strings, see http://go.microsoft.com/fwlink/?LinkId=723263.
         => optionsBuilder.UseNpgsql("Host=nutrinova-db;Port=5432;Database=nutrinovadb;Username=admin;Password=Pleasegivemecoke!");
```
Comment (garv5014) [inline nutrinova-api/NutrinovaData/NutrinovaDbContext.cs:54]: Should we pull these out into a secret?

File: nutrinova-ui/src/app/test/page.tsx [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default function Test() {
+  return <div>Hello</div>;
```
Comment (garv5014) [inline nutrinova-ui/src/app/test/page.tsx:2]: Does this file still need to be here?

Comment (Cwighty) [reply nutrinova-ui/src/app/test/page.tsx:2]: no

File: nutrinova-api/NutrinovaData/NutrinovaDbContext.cs (hunk 2 above)
Comment (Cwighty) [reply nutrinova-api/NutrinovaData/NutrinovaDbContext.cs:54]: eventually, right now its just for our development db

File: nutrinova-api/.editorconfig (hunk 1 above)
Comment (Cwighty) [reply nutrinova-api/.editorconfig:117]: done

**Review by andrewbengordon — APPROVED** [review]

**Pull request merged**